beacon-chain/blockchain/process_block.go

```diff
@@ -106,6 +107,33 @@ func (s *Service) onBlock(ctx context.Context, signed block.SignedBeaconBlock, b
 		return err
 	}
 
+	// If slasher is configured, forward the attestations in the block via
+	// an event feed for processing.
+	if features.Get().EnableSlasher {
+		// Feed the indexed attestation to slasher if enabled. This action
+		// is done in the background to avoid adding more load to this critical code path.
+		go func() {
+			ctx := context.TODO() // Shadow context as given this is a side-effect we don't want to affect the main thread.
```

Comment:
I don't think this comment is correct. The reason we are doing this is the same reason you gave on lines 124 and 125.

```
// Using a different context to prevent timeouts as this operation can be expensive
// and we want to avoid affecting the critical code path.
```

Reply:
and then you dont need that other context.TODO(), both functions can use `ctx`